packages/php/blueprint/src/Importers/ImportRunSql.php

```diff
@@ -79,4 +146,173 @@ public function check_step_capabilities( $schema ): bool {
 
 		return true;
 	}
+	/**
+	 * Check if the SQL query type is allowed.
+	 *
+	 * @param string $sql_content The SQL query to check.
+	 * @return bool True if the query type is allowed, false otherwise.
+	 */
+	private function is_allowed_query_type( string $sql_content ): bool {
+		$uppercase_sql_content = strtoupper( trim( $sql_content ) );
+
+		foreach ( self::ALLOWED_QUERY_TYPES as $query_type ) {
+			if ( 0 === stripos( $uppercase_sql_content, $query_type ) ) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Check for suspicious comment patterns that might hide malicious code.
+	 *
+	 * This method detects various types of SQL comments that might be used
+	 * to hide malicious SQL commands or bypass security filters.
+	 *
+	 * @param string $sql_content The SQL query to check.
+	 * @return bool True if suspicious comments found, false otherwise.
+	 */
+	private function contains_suspicious_comments( string $sql_content ): bool {
+		// Quick check if there are any comments at all before running regex.
+		if ( strpos( $sql_content, '--' ) === false &&
+		strpos( $sql_content, '/*' ) === false &&
+		strpos( $sql_content, '#' ) === false ) {
```

Comment:
Some formatting issues here

Reply:
Reformatted in d9ad44f2045eed0921f9be733d34bc8bae90d347